### PR TITLE
E2E: Improve the setup script

### LINKED ISSUE #9646: Playwright E2E Tests: Automating the loading of the *.txt files
Currently, for each .txt file that is inside the folder `bin`, there is a specific command to save that template in WordPress ([source code](https://github.com/woocommerce/woocommerce-blocks/blob/90760e4be19c5e719d8ae20fcb372d01ce63c191/tests/e2e-pw/bin/test-env-setup.sh/#L113-L153)). Now, it is easy to maintain because we have few .txt files, but it will not be when there are more. So, we should create a dedicated script that automatically inserts all the posts/pages reading all the .txt. 

Also, should we continue to use `.txt` as extension? If we use the `.html`, we could use the syntax highlight of the code editors.

### LINKED ISSUE #10594: E2E: improve environment setup process
## Is your feature request related to a problem? Please describe.
Currently, it takes ≈ 4 minutes to set up our test environments. 

## Describe the solution you'd like
We should cut this duration down as much as possible to get faster feedback and save CI minutes.

### PR DESCRIPTION
**Pull request by dinhtungdu**
This PR are trying to some related issues at once. I'm sorry about that. In this PR:

- We run the setup script inside the container instead of sending commands through docker sequentially.
- We break the setup script into smaller scripts and run them in parallel.
- We automate loading the tests posts.
- We change the extensions of test posts and pages from `txt` to `html` to use syntax highlighting in the editor.
- We make the setup script more scaleable. In the future:
  - If we need more test posts, drop the html file containing the post content into `tests/e2e/bin/posts/` and and the setup script will pick it automatically.
  - If we need more configuration, create a new file inside `tests/e2e/bin/scripts/` and the setup script will pick it automatically.

By introducing all of those changes above, we're able to cut the time running `npm run evn:restart` down to `49s` on my machine, while the previous setup time is `2m41s` in my recent test.

On CI, the improvement is also notable.

image

Fixes #9646, fixes #10594 

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

#### User Facing Testing

Ensure the Playwright tests are still passing.

* [x] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [x] Feature plugin
* [ ] Experimental

### Performance Impact

Improve the speed of the provision process for e2e environment.

### Changelog

N/a

### CODE REVIEW

File: tests/e2e/bin/scripts/index.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Empty the site to remove unused pages and posts created by WP and Woo.
+wp site empty --yes
+
+# Attributes must be created before importing products.
+bash $script_dir/attributes.sh
+
+# Run all scripts in parallel at maximum 10 at a time
+find $script_dir/parallel/*.sh -maxdepth 1 -type f | xargs -P10 -n1 bash
```
Comment (gigitux) [inline tests/e2e/bin/scripts/index.sh:12]: Nice 🚀

File: tests/e2e/bin/scripts/parallel/pages.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && cd ../../pages && pwd)"
+
+wp post create \
+	--menu_order=0 \
+	--post_type=page \
+	--post_status=publish \
+	--post_author=1 \
+	--post_title='All Products block' \
+	${script_dir}/all-products.html
```
Comment (gigitux) [inline tests/e2e/bin/scripts/parallel/pages.sh]: Is it still necessary?

**Review by gigitux — APPROVED** [review]
Thanks @dinhtungdu, for your effort in improving E2E test infrastructure! Great work and LGTM!

I just added a comment because I'm not sure that `all products page` is necessary, but I'm pre-approving the PR.

File: tests/e2e/bin/scripts/parallel/pages.sh (hunk 2 above)
Comment (dinhtungdu) [reply tests/e2e/bin/scripts/parallel/pages.sh]: I'm not sure. In this PR, I'm trying to convert the old script to the new one only. Searching our tests, look like we aren't using that page anywhere.

Comment (nielslange) [reply tests/e2e/bin/scripts/parallel/pages.sh]: @gigitux and @dinhtungdu In https://github.com/nielslange/woo-test-environment, the `All Products block` is used for the shop page, see https://github.com/nielslange/woo-test-environment/blob/5cfa23af4a4aa426f71b7a626ad8a3786f722ccb/command.php#L212-L216

I'll try to review this PR later today and will double-check if the `All Products block` is needed or not.

Comment (nielslange) [reply tests/e2e/bin/scripts/parallel/pages.sh]: @gigitux and @dinhtungdu These are the only places, where we interact with the `All Products block`:

https://github.com/woocommerce/woocommerce-blocks/blob/1d0af208720498af118e76f9e0e3c641f708b1e0/tests/e2e/tests/price-filter/price-filter.block_theme.side_effects.spec.ts#L31 

https://github.com/woocommerce/woocommerce-blocks/blob/1d0af208720498af118e76f9e0e3c641f708b1e0/tests/e2e/tests/price-filter/price-filter.block_theme.side_effects.spec.ts#L46-L48

https://github.com/woocommerce/woocommerce-blocks/blob/1d0af208720498af118e76f9e0e3c641f708b1e0/tests/e2e/tests/price-filter/price-filter.block_theme.side_effects.spec.ts#L77-L79

https://github.com/woocommerce/woocommerce-blocks/blob/1d0af208720498af118e76f9e0e3c641f708b1e0/tests/e2e/tests/mini-cart/mini-cart.block_theme.spec.ts#L113

As far as I can tell, the page `All Products block` is not loaded directly anywhere and can safely be removed.

**Review by nielslange — APPROVED** [review]
Thanks for working on this, @dinhtungdu. The environment starts indeed much faster. Well done that you separated the setup steps into various shell scripts. 👏

File: tests/e2e/bin/scripts/parallel/pages.sh (hunk 2 above)
Comment (dinhtungdu) [reply tests/e2e/bin/scripts/parallel/pages.sh]: I removed the All Products block page in https://github.com/woocommerce/woocommerce-blocks/pull/10611/commits/4473ffcba170d7363b293280ed1254cde7ef7fc0

**Pull request merged**